src/components/AllMovies/AllMovies.js

```diff
@@ -0,0 +1,47 @@
+import './AllMovies.css';
+import Card from '../Card/Card';
+import FilterRibbon from './FilterRibbon/FilterRibbon';
+import PropTypes from 'prop-types';
+import Loader from '../LoadSpinner/LoadSpinner';
+
+const AllMovies = ({  waitingForFetch, serverError, movies, changeSearch, searchValue  }) => {
+  let movieCards;
+
+  if(movies) {
+    movieCards = movies.map(movie => {
+      return (
+        <Card
+          movieTitle={movie.title}
+          moviePoster={movie.poster_path}
+          movieYear={movie.release_date.slice(0,4)}
+          key={movie.id}
+          id={movie.id}
+        />
+      )
+    })
+  }
```

Comment:
Some debate, but also could be pulled out into its own Movie component. AllMovies renders Movies.

Reply:
Sorry, I sort of don't follow.  I have <Card /> as the solo component. I iterate through the array of movies to create the Card component for them all and generate the component code for `allMovies`... Where would the `Movie` component fit in (I might be missing something but I feel like I'm doing that with card?)

Reply:
My comment is more around semantics. Like a Movies component renders a component called Movie and Movie renders a component named Card. What you did was perfectly okay, but naming is a big part of our work to help other devs understand our code.

Reply:
Maybe its a MovieCard....

Semantics....man they never end.

Reply:
One more thing, if you were to check for movies and if they dont exist, throw up the loader like `if (movies) return <Loader />;`

then

That whole `let movies`; and  `if(movie)` conditional would not need to be there. 

But I would put it behind a method.

```suggestion
 const showMovies = () => {
     return movies.map(movie => {
         return (
            <Card
              movieTitle={movie.title}
              moviePoster={movie.poster_path}
              movieYear={movie.release_date.slice(0,4)}
              key={movie.id}
              id={movie.id}
            />
          )
        })
      }
  }
```